bitcoin/src/taproot.rs

```diff
@@ -987,6 +1004,46 @@ impl LeafNode {
     }
 }
 
+/// Script leaf node in a taproot tree along with the merkle proof to get this node.
+/// Returned by [`TapTree::script_leaves`]
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ScriptLeaf<'leaf> {
+    /// The version of the script leaf.
+    version: LeafVersion,
+    /// The script.
+    script: &'leaf Script,
+    /// The merkle proof (hashing partners) to get this node.
+    merkle_branch: TaprootMerkleBranch,
```

Comment:
Shouldn't this be borrowed as well?

Reply:
Indeed, I don't know what I was thinking here.